build/images/base/Dockerfile

```diff
@@ -13,6 +13,31 @@
 # limitations under the License.
 
 ARG BUILD_TAG
+
+FROM ubuntu:22.04 AS gobgp-binary
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends wget ca-certificates golang
+
+ENV PATH=$PATH:/usr/local/go/bin
+
+WORKDIR /tmp/gobgp
+
+RUN go mod init temp-gobgp && \
+    go get github.com/osrg/gobgp/v3
```

Comment:
If `go` is needed, use `FROM golang:${GO_VERSION}` for the base image, not `ubuntu:22.04`
The image already includes other tools you need such as `dpkg`, `wget` and `tar`
We never install golang from the apt repos.

However, I can see from this that getting the correct version of gobgp from the Antrea go.mod is not going to be possible from the Docker build itself. We would need access to the go.mod file itself, which is not practical because the file is not part of the build context.
The current code here is not correct because we are not actually getting the gobgp version used by Antrea. We are getting the latest minor version of gobgp v3.

So instead we should have the following hybrid approach:
- Introduce the `GOBGP_VERSION` docker ARG
- From the build.sh script, use `grep "github.com/osrg/gobgp" ../../../go.mod | head -n1 | cut -d " " -f2 | cut -c 2-` to get the correct version, and use the value to set `GOBGP_VERSION` for the docker build

With this approach, you can keep using `ubuntu:22.04` as the base image

Reply:
Right, we should same gobgp used in Antrea